### PR TITLE
[WNMGDS-2161] Compact pagination

### PR DESCRIPTION
**Pull request by zarahzachz**
## Summary

WNMGDS-2161

This work resolves a lot of a11y issues with Pagination:
- Page now reads `aria-current="page"`
- Added a heading for AT that announces the Pagination label + current page of total pages whenever a page changes
- Consolidated the disabled nav links (rendered as `span`) and button links to resolve issue of aria not being allowed on `span` tags

## Test

View [demo link](https://cmsgov.github.io/design-system/branch/WNMGDS-2161/compact-pagination/components/pagination/?theme=core) or run locally, navigate using Mac VO (for all 3 Pagination examples)

- [Default Pagination example](https://cmsgov.github.io/design-system/branch/WNMGDS-2161/compact-pagination/storybook/iframe.html?globals=theme:core&id=components-pagination--default&viewMode=story)
- [Hidden nav example](https://cmsgov.github.io/design-system/branch/WNMGDS-2161/compact-pagination/storybook/iframe.html?globals=theme:core&args=&id=components-pagination--hidden-navigation&viewMode=story)
- [Compact example](https://cmsgov.github.io/design-system/branch/WNMGDS-2161/compact-pagination/storybook/iframe.html?globals=theme:core&args=&id=components-pagination--compact-navigation&viewMode=story)

### CODE REVIEW

**Review by pwolfert — APPROVED** [review]
It sounds really nice with VoiceOver (to me, not an expert). Good work! A couple questions below but everything else LGTM.

File: packages/design-system/src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -247,33 +247,32 @@ function Pagination({
...
-          style={{ visibility: isNavigationHidden ? 'hidden' : 'visible' }}
-          aria-hidden={isNavigationHidden}
-        >
-          <span className="ds-c-pagination__nav--img-container ds-c-pagination__nav--img-container-previous">
-            {startIcon}
-          </span>
-          {startLabelText ?? t('pagination.startLabelText')}
+    <nav className={classes} aria-labelledby="pagination-heading" {...rest}>
+      <span aria-live="polite" role="status" className="ds-u-visibility--screen-reader">
+        <h2 id="pagination-heading">
```
[the remark this answers is not in the document]
Comment (zarahzachz) [reply packages/design-system/src/components/Pagination/Pagination.tsx:252]: I considered that, but given Pagination is typically used as a higher level navigation element, it made sense to hard code it to an H2. I imagine all navigational landmark regions should have a higher heading level just for quick access via the rotor, but I could be wrong. @SparrowSpit what are your thoughts?

**Review by forrestbaer — APPROVED** [review]
Looks good!

File: packages/design-system/src/styles/components/_Pagination.scss [hunk 2]
```diff
@@ -17,7 +18,6 @@
   }
   li {
     display: inline-block;
```
[the remark this answers is not in the document]
Comment (zarahzachz) [reply packages/design-system/src/styles/components/_Pagination.scss:20]: i like this change. i wonder if we need to consider a similar vertical positioning adjustment for the prev/next buttons? they do sit pretty low compared to the rest of the page links.

[the remark this answers is not in the document]
Comment (zarahzachz) [reply packages/design-system/src/styles/components/_Pagination.scss:20]: Created WNMGDS-2211 to address these changes

**Pull request merged**